io/eolearn/io/local_io.py

```diff
@@ -389,16 +404,24 @@ def execute(self, eopatch=None, *, filename=None):
         filesystem, filename_paths = self._get_filesystem_and_paths(filename, eopatch.timestamp, create_paths=False)
 
         with filesystem:
-            with filesystem.openbin(filename_paths[0], 'r') as file_handle:
-                with rasterio.open(file_handle) as source:
+            for path in filename_paths:
+                if not filesystem.exists(path):
+                    raise FileNotFoundError(f'The file "{path}" to import from tiff does not exist.')
```

Comment:
Since today I'm making a new release I'll merge this PR and just remove these 3 lines to avoid making unnecessary requests to AWS. But feel free to continue the discussion or open a new PR regarding this.

Reply:
Hm, interesting. Thanks for providing the benchmark, I understand your reasoning now. 

However, I believe we should try to optimize the task for the case where the task execution succeeds without errors. In that case, the task should take as little time as possible and make as few requests to AWS bucket as possible. It is expected that a user will ensure that all GeoTiffs are available on the bucket before running the task. Therefore, a case, where an image is missing, should be treated as a special case which can happen only occasionaly. I think it is ok if it takes `2s` more to fail.

If someone would have a use-case where a lot of images would be missing on the bucket I would recommend that such `filesystem.exists` checks are run before `ImportFromTiff` task is executed.

Reply:
Yeah I forgot to formally submit the review, but I had already performed the test last week, my bad. But yeah I agree with your conclusions anyhow.